code/tailer/datastripe.q

```diff
@@ -0,0 +1,196 @@
+
+    /- get process specific taildir location
+    basedir:` sv dir,.proc.procname;
+    dir:` sv basedir,`$ string .wdb.currentpartition;
+
+    /- get symbol enumeration
+    partitionint:`$string (where s=value [`.]keycol)0;
+
+    /- create directory location for selected partition
+    directory:` sv .Q.par[dir;partitionint;tablename],`;
+    .lg.o[`save;"Saving ",string[s]," data from ",string[tablename]," table to partition ",string[partitionint],". Table contains ",string[count enumdata]," rows."];
+    .lg.o[`save;"Saving data down to ",string[directory]];
+
+    /- upsert select data matched on partition to specific directory
+    .[upsert;
+        (directory;enumdata);
+        {[e] .lg.e[`upserttopartition;"Failed to save table to disk : ",e];'e}
+    ];
+
+    /- check if sym file exists in taildir, create symlink to HDB sym file if not
+    if[not `sym in key hsym basedir;createsymlink[basedir;.wdb.hdbdir;`sym]];
+    };
+
+savetables:{[dir;tablename]
+    /- function to get keycol for table from access table
+    keycol:`sym^.wdb.tablekeycols tablename;
+
+    /- get distinct values to partition table on
+    partitionlist: ?[tablename;();();(distinct;keycol)];
+
+    /- enumerate and then split by keycol
+    symdir:` sv dir,.proc.procname;
+    enumkeycol: .Q.en[symdir;value tablename];
+    splitkeycol: {[enumkeycol;keycol;s] ?[enumkeycol;enlist (=;keycol;enlist s);0b;()]}[enumkeycol;keycol;] each partitionlist;
+
+    /-upsert table to partition
+    upserttopartition[dir;tablename;keycol;] each splitkeycol where 0<count each splitkeycol; 
+    
+    /- run a garbage collection (if enabled)
+    .gc.run[];
+    };
+
+savealltables:{[dir]
+    /- function takes the tailer hdb directory handle and a timestamp
+    /- saves each table up to given period to their respective partitions
+    savetables[dir;]each .wdb.tablelist[];
+
+    /- delete data that has been saved
+    @[`.;;0#] each .wdb.tablelist[];
+
+    /- trigger reload of access tables and intradayDBs in all tail reader processes
+    .tailer.dotailreload[`]};
+
+savedownfilter:{[dir]
+    /- checks each table in memory against a size threshold
+    /- saves any tables above that threshold
+    totals:{count value x}each .wdb.tablelist[];
+    /- log and return from function early if no table has crossed threshold
+    if[all totals<.wdb.numrows;
+        .lg.o[`save;"No tables above threshold, no tables saved"];
+        :();
+    ];
+
+    /- log and savedown any tables above threshold
+    .lg.o[`save;"Saving ",(", " sv string tabstosave:.wdb.tablelist[] where totals>.wdb.numrows)," table(s)"];
+    savetables[dir;]each tabstosave;
+
+    /- delete data that has been saved
+    @[`.;;0#] each tabstosave;
+
+    /- trigger reload of access tables and intradayDBs in all tail reader processes
+    .tailer.dotailreload[`]
+    };
+
+/- Timer to repeat savealltables with period defined in tailer.q settings
```

Comment:
```suggestion
/- Timer to call savedownfilter with period defined in tailer.q settings
```

Reply:
@jonathonmcmurray This change looks good to me but I don't have permissions to commit it

Reply:
applied